model/compile/QtiAssetCompiler/QtiItemAssetCompiler.php

```diff
@@ -49,29 +58,43 @@ public function extractAndCopyAssetFiles(Item $qtiItem, Directory $publicDirecto
         $xincludeLoader->load();
 
         $packedAssets = [];
-
         foreach ($assetParser->extract() as $type => $assets) {
             foreach ($assets as $key => $assetUrl) {
                 if ($this->isBlacklisted($assetUrl)) {
                     continue;
                 }
-
                 $packedAsset = $this->resolve($resolver, $assetUrl, $type);
-
                 $replacement = $this->getReplacementName($packedAsset);
                 $packedAsset->setReplacedBy($replacement);
 
                 if ($type != 'xinclude') {
-                    $this->copyAssetFileToPublicDirectory($publicDirectory, $packedAsset);
+                    if (!$this->replaceToExternalSource($packedAsset, $qtiItem)) {
+                        $this->copyAssetFileToPublicDirectory($publicDirectory, $packedAsset);
+                    }
                 }
-
                 $packedAssets[$assetUrl] = $packedAsset;
             }
         }
 
         return $packedAssets;
     }
 
+    private function replaceToExternalSource(PackedAsset &$packedAsset, Item $qtiItem): bool
```

Comment:
I don't think the reference is necessary as we are dealing with an object for `$packedAsset`